github/githubapp.go

```diff
@@ -0,0 +1,109 @@
+package github
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"path"
+	"strconv"
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+)
+
+type GitHubApp struct {
+	ClientId       string `json:"client_id"`
+	InstallationId int64  `json:"installation_id"`
+	PrivateKey     string `json:"private_key"` // SENSITIVE
+}
+
+func (app *GitHubApp) IsZero() bool {
+	if app.ClientId != "" {
+		return false
+	}
+	if app.InstallationId != 0 {
+		return false
+	}
+	if app.PrivateKey != "" {
+		return false
+	}
+	return true
+}
+
+// generateJWTtoken returns a signed JWT token used to authenticate as GitHub App
+func generateJWTtoken(clientId, privateKey string) (string, error) {
```

Comment:
if you pass the current time as a parameter, then this function is easily testable stand-alone.

Reply:
Then I have to expose it to be able to test it...

Test package name: `package github_test`

Reply:
Ah this is not so simple... jwt package has some internal timers...